doc: known issues: Add bluetooth known issues for 1.4.0 (#3259)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

|BLE|
=====

.. rst-class:: v1-4-0 v1-3-2 v1-3-1 v1-3-0

NCSDK-6844: Security procedure failure can terminate GATT client request

>   A security procedure terminates the current GATT client request that is currently in progress, unless the current request was the reason to initiate the security procedure.

Reviewer: That are a lot of currents in there now ... ;)
Either
```suggestion
  A security procedure terminates the current GATT client request, unless the current request was the reason to initiate the security procedure.
```
or 
```suggestion
  A security procedure terminates the GATT client request that is currently in progress, unless the current request was the reason to initiate the security procedure.
```

Author: Removed 2x "current", the second one seemed redundant to me as well.

Other issues
------------

.. rst-class:: v1-4-0 v1-3-2 v1-3-1 v1-3-0

NCSDK-6844: Security procedure failure can terminate GATT client request

>   Security procedure terminates the current GATT client request in progress when this request was not the reason to initiate the security procedure.
>   If a new GATT client request is queued then this can now potentially cause a GATT transaction violation and could fail as well.
> 
>   **Workaround:** Don't initiate security procedure in parallel with GATT client requests.

Reviewer: ```suggestion
  A security procedure terminates the GATT client request that is currently in progress, unless the current request was the reason to initiate the security procedure.
  If a new GATT client request is queued at this time, this might potentially cause a GATT transaction violation and fail as well.

  **Workaround:** Do not initiate a security procedure in parallel with GATT client requests.
```

Other issues
------------

.. rst-class:: v1-4-0 v1-3-2 v1-3-1 v1-3-0

> NCSDK-6844: Security procedure failure can terminate GATT client request

Reviewer 2: Is it the case that the GATT client request comes from the same side that is initiating the security procedure?

Author: It does not have to be, the trigger is the encrypt change callback to ATT with an error code.
This originates from bt_conn_security_changed in the code. For example the HCI encrypt change event.

In this case it was triggered by SMP so that the security_changed callback could be given to the application since it initiated security.

```
	if (atomic_test_bit(smp->flags, SMP_FLAG_SEC_REQ)) {
		/* Notify higher level that security failed if security was
		 * initiated by slave.
		 */
		bt_conn_security_changed(conn, BT_HCI_ERR_PIN_OR_KEY_MISSING,
					 BT_SECURITY_ERR_PIN_OR_KEY_MISSING);
	}
```

  Security procedure terminates the current GATT client request in progress when this request was not the reason to initiate the security procedure.
  If a new GATT client request is queued then this can now potentially cause a GATT transaction violation and could fail as well.

  **Workaround:** Don't initiate security procedure in parallel with GATT client requests.

> .. rst-class:: v1-4-0

Reviewer 2: NCSDK-6872. Is that only for NCS 1.4.0?

Author: Removed it, but don't know which releases are affected yet.
